libr/anal/fcn.c

```diff
@@ -328,16 +328,14 @@ R_API void r_anal_fcn_free(void *_fcn) {
 	free (fcn);
 }
 
-// TODO: use this function if anal.jmpmid == false
-static RAnalBlock *bbget(RAnalFunction *fcn, ut64 addr) {
+static RAnalBlock *bbget(RAnalFunction *fcn, ut64 addr, bool jumpmid) {
 	RListIter *iter;
 	RAnalBlock *bb;
 	r_list_foreach (fcn->bbs, iter, bb) {
 		ut64 eaddr = bb->addr + bb->size;
-		if (bb->addr >= eaddr && addr == bb->addr) {
-			return bb;
-		}
-		if ((addr >= bb->addr) && (addr < eaddr)) {
+		if (((bb->addr >= eaddr && addr == bb->addr)
```

Comment:
ok c5d5a29bf319d710f0ce508493e1e552f02c72ca

> ... in both cases, I think there shouldn't be any check here

There might be some (buggy) arch code that forgot to set bb->size (and btw bb->size is an `int` so it can be negative), and I don't know enough to remove the check.

Reply:
it's ok. as said, it's not strictly related to this PR but this check is either useless or it is a sign of something wrong ;)

Reply:
Ah, the way we can solve these "might be some buggy arch code.." nowadays is by using `r_return_val_if` or `r_warn_if_reached` or `r_warn_if_fail`. They are good checks for things that we assume should always be true but may not be in case of some programming errors.

Reply:
yeah, random comment indeed ;)